### DISCUSSION TITLE
MinerU is an awesome but very slow

### DISCUSSION THREAD

**Discussion by sleepingcat4** [magicpdf/Magic-PDF#1226] [General]
I had been using MinerU and I realised it was painfully slow. Taking almost 15 minutes on 3090 to process a 44 page 1 pdf. That I don't think feasible if someone wants to use it for large scale. is there any test being done with MinerU to parse pdf and take benchmark?

also any test being done for H100s or A100 GPUs for the library?

**Comment by myhloli** [comment]
Your speed is clearly not normal. Our online demos all use A10 GPUs and can achieve approximately 1-2 pages per second. Please contact you to check if the system environment has correctly enabled GPU acceleration.

**Comment by hongbo-miao** [comment]
Just provide a data point, for me, I tested on Ubuntu with NVIDIA RTX A2000 8GB Laptop GPU. It can process 1 page every 5 seconds.
I set up based on https://github.com/opendatalab/MinerU/discussions/1374

**Comment by sleepingcat4** [comment]
@hongbo-miao thanks for the data-point. I am using a CPU native non-AI parser that can parse in 100 miliseconds a single 44 page pdf. But, I think I can tweak MineU later to get it down to 10 seconds for single 44-page pdf.

**Reply by gaddam-thrinath** [reply]
can you plz the process  to setup for CPU native non-AI parser
